packages/tx/src/EIP2930Transaction.ts

```diff
@@ -0,0 +1,385 @@
+import {
+  Address,
+  BN,
+  bnToHex,
+  bnToRlp,
+  bufferToHex,
+  ecrecover,
+  keccak256,
+  rlp,
+  setLengthLeft,
+  toBuffer,
+} from 'ethereumjs-util'
+import { BaseTransaction } from './baseTransaction'
+import {
+  AccessList,
+  AccessListBuffer,
+  AccessListItem,
+  isAccessList,
+  JsonTx,
+  TxData,
+  TxOptions,
+} from './types'
+
+// secp256k1n/2
+const N_DIV_2 = new BN('7fffffffffffffffffffffffffffffff5d576e7357a4501ddfe92f46681b20a0', 16)
+
+type EIP2930ValuesArray = [
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  AccessListBuffer,
+  Buffer?,
+  Buffer?,
+  Buffer?
+]
+
+export class EIP2930Transaction extends BaseTransaction<EIP2930Transaction> {
```

Comment:
Should we rename this class to `AccessListTransaction`?

Reply:
Had to let this sink in a bit. I generally think that we should avoid naming things after EIPs since it is likely hard for users not having dealt with an EIP so intensely to just remember by number (I do have this very often myself). In this transaction case I wonder if we will always be able to find short descriptive (enough) names.

I think it is valuable to broaden the scope here to include [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) in the discussion. There we have 2-3 points:

- `AccessLists` are *also*  included in this transaction type (I actually wonder if we are not running into a whole lot of dependency hell with these tx types being so generic (so what tx type is depending on what other type and so on?), another discussion though)
- *This* (so: `EIP-1559`) is actually an EIP people very much *do* refer to by number

When counting these things in I currently therefore have the tendency to name by *both* the EIP and something descriptive, think this would give the best compromise of staying unambiguous and giving some hint on the functionality at the same time, or is this overblown? 🤔

Therefore some naming first-shots:

- `EIP2930AccessListTransaction`
-`EIP1559FeeMarketTransaction`

We could also shorten to:

- `EIP2930AccessListTx`
- `EIP1559FeeMarketTx`

(I think I would have some preference for the first since `Tx` is coming in so short on the second and gets lost a bit). But I have to say I am generally not completely settled yet on the whole subject.

Would think that @ryanio might have a good intuition here? Other opinions?

Reply:
I like it. Can we put the descriptive name first and then the EIP? So `AccessListTransactionEIP2930`, or `AccessListEIP2930Transaction` (I think I like the latter).

Reply:
Yeah, that's likely even better with the descriptive name first, and - also yes - would put `Transaction` last, this is otherwise confusing.

Maybe nevertheless wait for another voice here until we change.

Reply:
I like this too. Really good.

Reply:
great idea, descriptive name first seems friendly :)